src/io/files.js

```diff
@@ -683,6 +683,68 @@ p5.prototype.loadXML = function() {
   return ret;
 };
 
+/**
+ * Loads a binary file from a file or a URL, and returns an Object
+ * with a bytes property containing a Uint8Array.
+ *
+ * This method is asynchronous, meaning it may not finish before the next
+ * line in your sketch is executed.
+ *
+ * @method loadBytes
+ * @param  {String}        path       name of the file or url to load
+ * @param  {function}      [callback] function to be executed after
+ *                                    loadBytes() completes, data is passed
+ *                                    in as first argument
+ * @param  {function}      [errorCallback] function to be executed if
+ *                                    there is an error, response is passed
+ *                                    in as first argument
+ * @return {Object}             contains a Uint8Array bytes property
+ * @example
+ *
+ * <p>Calling loadBytes() inside preload() guarantees to complete the
+ * operation before setup() and draw() are called.</p>
+ *
+ * <div><code>
+ * // TODO: Add Example with preload
+ * </code></div>
+ *
+ * <p>Outside of preload(), you may supply a callback function to handle the
+ * data:</p>
+ * <div><code>
+ * // TODO: Add Example with callback
+ * </code></div>
+ *
+ * @alt
+ * 50x50 ellipse that changes from black to white depending on the current humidity
+ * 50x50 ellipse that changes from black to white depending on the current humidity
+ *
+ */
+p5.prototype.loadBytes = function(path, callback, errorCallback) {
+  var self = this;
+  var ret = {};
+  var oReq = new XMLHttpRequest();
+  oReq.open('GET', path, true);
+  oReq.responseType = 'arraybuffer';
+  oReq.onload = function(oEvent) {
+    if (oReq.status >= 400) {
```

Comment:
In httpDo, it checks on fetch's `res.ok` which is only valid if the status code is 2xx. So `status >= 200 && status <300` would be the same behaviour as the rest of p5 but I'm not sure what the feeling for this is.

Reply:
Incase people come from the video and don't understand why `status >= 200 && status < 300` makes sense.

Here is some further reading:
https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/status

Reply:
A couple quick comments before I test the code locally: it would probably be best to check for `status >= 200 && status < 300` only as `>=300` will not return a response body and so it won't populate the intended variable. The client is supposed to take care of `>=300` but I'm guessing without knowledge of the endpoint we can't handle it properly.

As @meiamsome mentioned, if response is not ok an error should be thrown so that it won't fail silently, I'm not sure if in this case simply throwing the error will make the error be passed onto `oReq.onerror` otherwise the user won't really be able to handle the error themselves.